pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -192,6 +194,23 @@ targetHostCmd() {
     fi
 }
 
+targetHostSudoCmd() {
+    restoreMaybeSudo=
+    previousMaybeSudo=("${maybeSudo[@]}")
+    if [[ -n "$remoteSudo" ]]; then
+        maybeSudo=("${sudoCommand[@]}")
+        restoreMaybeSudo=1
+    fi
+
+    targetHostCmd "$@"
+    ret=$?
+
+    if [[ -n "$restoreMaybeSudo" ]]; then
+        maybeSudo=("${previousMaybeSudo[@]}")
+    fi
+    return $ret
+}
```

Comment:
This function is really hard to follow.
I think it's ok to duplicate the implementation of `targetHostCmd`. It's only five lines, and it'd be so much easier to understand that it's also easier to change coherently.

Reply:
I've rewritten the function to be a lot simpler :+1: